recipes/zip/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class ZipConan(ConanFile):
+    name = "zip"
+    license = "Unlicense"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kuba--/zip"
+    description = "A portable, simple zip library written in C"
+    topics = ("conan", 'zip', 'compression', 'c', 'miniz', 'portable', 'hacking')
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake",
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["CMAKE_DISABLE_TESTING"] = True
+        cmake.configure()
```

Comment:
Or we can remove https://github.com/conan-io/conan-center-index/pull/5081/files#diff-c5d00f02f2f77b30176b2669e7f269506265601816d1c0dc277900fdcf0408adR25-R27

```suggestion
        cmake.configure(build_folder=self._build_subfolder)
```

Reply:
well, not sure if it improves anything, doesn't seem to be required (builds just fine without this line), but only makes code longer -> harder to read and understand

Reply:
It helps for development, separated build cache from copied sources.

Reply:
maybe it does, but it fails to build such way for me, so please don't suggest it, it doesn't work